src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -431,33 +423,29 @@ header_cb(char *data, size_t size, size_t nmemb, void *connp)
         param.type = SOL_HTTP_PARAM_COOKIE;
         cookie_name_size = 0;
         while (sep[cookie_name_size++] != '=') ;
-        param.value.key_value.key = curl_easy_unescape(connection->curl,
-            sep, cookie_name_size - 1, NULL);
+        param.value.key_value.key.data = sep;
```

Comment:
is this right? AFAIU you got a raw cookie from CURL, after all you had to find "=" in it, we're keeping parameters unescaped... then we should have to unescape here, no?